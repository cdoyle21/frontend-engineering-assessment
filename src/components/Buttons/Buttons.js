import React from 'react';
import Aux from '../../hoc/Aux';
import classes from './Buttons.module.css';

const buttons = ( props ) => (
    <Aux>
        <button onClick={props.clicked} className={classes.Buttons} value="button test">
            {props.children}
        </button>
    </Aux>
)

export default buttons;