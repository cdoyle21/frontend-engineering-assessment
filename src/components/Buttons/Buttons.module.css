.Buttons {
    background-color: #e5fd3d;
    outline: none;
    cursor: pointer;
    border: 1px solid #612efb;
    color: #612efb;
    font-family: inherit;
    font-size: 1.2em;
    padding: 15px 30px;
    box-shadow: 2px 2px 2px #612efb;
}

.Buttons:hover, .Buttons:active {
    background-color: #A0DB41;
    border: 1px solid #612efb;
    color: #612efb;
}

.Buttons:disabled {
    background-color: #C7C6C6;
    cursor: not-allowed;
    border: 1px solid #ccc;
    color: #888888;
}

.Buttons:not(:disabled) {
    animation: enable 0.3s linear;
}