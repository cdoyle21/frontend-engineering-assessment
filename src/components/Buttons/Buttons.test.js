import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import React from 'react';
import Buttons from './Buttons';

configure({ adapter: new Adapter() });

describe('Render Buttons component', () => {
    const mockFn = jest.fn();
    it('should be defined', () => {
        expect(Buttons).toBeDefined();
      });
    it('should render Buttons component correctly', () => {
        const wrapper = shallow(
        <Buttons name="button" handleClick={mockFn} />,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should have a button value', () => {
        const wrapper = shallow(
            <Buttons name="button test" handleClick={mockFn} />
        );
        expect(typeof(wrapper.find('.Button').node.props.value)).toBe('string');
        expect(wrapper.find('.Button').node.props.value).toEqual('button test');
    });
    it('should have a click function', () => {
        const wrapper = shallow(
          <Buttons name='button' handleClick={mockFn} />
        );
        wrapper.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
  });
