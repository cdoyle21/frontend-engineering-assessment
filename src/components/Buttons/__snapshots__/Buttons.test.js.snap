// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Buttons component should render Buttons component correctly 1`] = `ShallowWrapper {}`;
